Classes/Plugin/Eid/SearchInDocument.php

```diff
@@ -38,29 +38,48 @@ class SearchInDocument
      */
     public function main(ServerRequestInterface $request)
     {
+        $output = [
+            'documents' => [],
+            'numFound' => 0
+        ];
+        // Get input parameters and decrypt core name.
         $parameters = $request->getParsedBody();
-        $encrypted = (string)$parameters['encrypted'];
-        $hashed = (string)$parameters['hashed'];
+        $encrypted = (string) $parameters['encrypted'];
+        $hashed = (string) $parameters['hashed'];
+        $count = intval($parameters['start']);
         if (empty($encrypted) || empty($hashed)) {
             throw new \InvalidArgumentException('No valid parameter passed!', 1580585079);
         }
         $core = Helper::decrypt($encrypted, $hashed);
-
-        $output = '';
-        if (!empty($core)) {
-            $query = (string)$parameters['q'];
-            $uid = (string)$parameters['uid'];
-            $start = (string)$parameters['start'];
-            $url = trim(Solr::getSolrUrl($core), '/') . '/select?wt=json&q=fulltext:(' . Solr::escapeQuery($query) . ')%20AND%20uid:' . $uid
-                . '&hl=on&hl.fl=fulltext&fl=uid,id,page&hl.method=fastVector'
-                . '&start=' . $start . '&rows=20';
-            $output = GeneralUtility::getUrl($url);
+        // Perform Solr query.
+        $solr = Solr::getInstance($core);
+        if ($solr->ready) {
+            $query = $solr->service->createSelect();
+            $query->setFields(['id', 'uid', 'page']);
+            $query->setQuery('fulltext:(' . Solr::escapeQuery((string) $parameters['q']) . ') AND uid:' . intval($parameters['uid']));
+            $query->setStart($count)->setRows(20);
+            $hl = $query->getHighlighting();
+            $hl->setFields('fulltext');
+            $hl->setMethod('fastVector');
```

Comment:
This method exists only from solarium 6.0.2.
Maybe, before it was possible with `$hl->setUsefastvectorhighlighter(true);`? Could'n test this because of non-ready solr above.

Reply:
You're right! The method is called `setUseFastVectorHighlighter()` in Solarium <6.x.

Reply:
This works now.